File: nodejs/http1mb.js
var http = require('http');
var bytes = new Buffer(1024*1024);

// start this file with `node http1mb.js` and then benchmark it:
// ab -c 100 -n 10000 http://127.0.0.1:8000/

for (var i = 0; i < bytes.length; i++) {
  bytes[i] = 100;
}

http.createServer(function (req, res) {
  res.writeHead(200);
  res.write(bytes);
  res.end();
}).listen(8000);

File: go/http1mb.go
// run like: go run http1mb.go
// and then benchmark it with:
// ab -c 100 -n 10000 http://127.0.0.1:8000/
package main

import "net/http"

func main() {
    bytes := make([]byte, 1024*1024)
    for i := 0; i < len(bytes); i++ {
        bytes[i] = 100
    }

    http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
        w.Write(bytes)
    })
    http.ListenAndServe(":8000", nil)
}
